Asan: Disable asan for CentOS and use statically link if possible.

chen-guanghua opened the pull request:
disable asan for centos7, fix #3347

winlinvip commented on trunk/auto/options.sh:
```diff
@@ -144,6 +144,11 @@ function apply_system_options() {
     fi
+
+    # Set centos version
+    if [[ $OS_IS_CENTOS == YES ]]; then
```
No need to check the version, let's disable Asan by default for all CentOS. But simply turning it off is not enough, we still need to support users to enable it themselves.

First, mark SRS_SANITIZER as RESERVED, and then set the initial value based on different systems.

Additionally, by default, enable asan static linking if possible.

`TRANS_BY_GPT3`

winlinvip commented on trunk/configure:
```diff
@@ -197,11 +197,6 @@ if [[ $SRS_STATIC == YES ]]; then
 
-# For asan(Google Address Sanitizer)
-if [[ $SRS_SANITIZER == YES && $OS_IS_X86_64 == YES && $SRS_SANITIZER_STATIC == YES ]]; then
-    SrsLinkOptions="${SrsLinkOptions} -static-libasan";
```
Why remove this link option?

`TRANS_BY_GPT3`

chen-guanghua replied:
Repeated at line 217

`TRANS_BY_GPT3`

winlinvip commented on trunk/auto/options.sh:
```diff
@@ -512,6 +512,13 @@ function apply_auto_options() {
+    # @see https://github.com/ossrs/srs/issues/3347
+    if [[ $SRS_SANITIZER == RESERVED && $OS_IS_CENTOS != YES && $OS_IS_X86_64 == YES ]]; then
+        echo "Disable asan for Centos"
+        SRS_SANITIZER=YES
```
I see the messages and comments, all of them are about disabling ASAN, why is it YES here?

Also, should we stop checking `OS_IS_X86_64`?

`TRANS_BY_GPT3`

chen-guanghua replied:
By default, it is RESERVED. If it is not CentOS, set it as SRS_SANITIZER=YES.

`TRANS_BY_GPT3`

winlinvip commented on trunk/auto/options.sh:
```diff
@@ -506,6 +506,13 @@ function apply_auto_options() {
+    # @see https://github.com/ossrs/srs/issues/3347
+    if [[ $SRS_SANITIZER == RESERVED && $OS_IS_CENTOS != YES ]]; then
+        echo "Enable asan by auto options."
+        SRS_SANITIZER=YES
```
If it is CentOS, then `SRS_SANITIZER=RESERVED` is actually not very good.

It should be explicitly set to NO.

`TRANS_BY_GPT3`

chen-guanghua committed:
Asan: Disable asan for CentOS and use statically link if possible. v5.0.127 (#3347)

1. Disable asan for all CentOS by default, however user could enable it.
2. Link asan statically if possible.

winlinvip committed:
Update version to v5.0.127
